flang/lib/Lower/Bridge.cpp

```diff
@@ -2659,8 +2730,11 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     localSymbols.clear();
   }
 
-  /// Instantiate the data from a BLOCK DATA unit.
-  void lowerBlockData(Fortran::lower::pft::BlockDataUnit &bdunit) {
+  /// Helper to generate GlobalOps when the builder is not positioned in a
+  /// any region block. This is required because the FirOpBuilder assumes it is
+  /// always positioned inside a region block.
+  void
+  createGlobalOutsideOfAnyContext(const std::function<void()> &createGlobals) {
```

Comment:
Doesn't creating the `fir::GlobalOp` and opening the building on that work? `GlobalOp` can have regions and entry blocks just like functions.

(This name is misleading. The bridge can't do anything at all without an MLIR context. Indeed, there is no such thing as "outside" that context. In fact, Globals have to be inside a ModuleOp or this is creating invalid IR.)